ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/Leaf.java

```diff
@@ -89,7 +89,12 @@ public String getMessage() {
         }
 
         // apply Network Actions on initial network
-        networkActions.forEach(na -> na.apply(network));
+        for (NetworkAction na : networkActions) {
+            boolean applicationSuccess = na.apply(network);
+            if (!applicationSuccess) {
+                throw new FaraoException(String.format("%s could not be applied on the network", na.getId()));
+            }
```

Comment:
this is quite violent :p

the exception is caught later one ?

Reply:
yes it is caught in the search tree